services/base.js

```diff
@@ -23,6 +25,12 @@ class BaseService {
     this._handleInternalErrors = handleInternalErrors;
   }
 
+  static render(props, namedParams, queryParams) {
```

Comment:
Can you give an example of when we need to pass `namedParams, queryParams` to `render()`? I don't think they get used anywhere in this PR.

Reply:
Number formatting, or work like in #1321 is an example. Would it be better to pass that stuff through `props`?

Reply:
I can see how if we went with the 'framework' approach, it would be useful to make these available here, but given we have to manually implement `render()`, construct all the stuff we want to pass it and call it ourselves, I think we should probably just explicitly pass what we need into `props`.